components/post_view/post_list/post_list.jsx

```diff
@@ -243,7 +243,7 @@ export default class PostList extends React.PureComponent {
             return;
         }
 
-        if (!this.state.olderPosts.allLoaded) {
+        if (!this.state.olderPosts.allLoaded && type === PostRequestTypes.BEFORE_ID) {
             const oldestPostId = this.getOldestVisiblePostId();
             await this.getPostsBefore(oldestPostId);
         } else if (!this.state.newerPosts.allLoaded) {
```

Comment:
Should this line be `} else if (!this.state.newerPosts.allLoaded && type !== PostRequestTypes.BEFORE_ID) {` to stop it from triggering on the case where type is `PostRequestTypes.BEFORE_ID` and `this.state.olderPosts.allLoaded` is true?

Reply:
The default value for type is `PostRequestTypes.BEFORE_ID` so it will be overridden only if either there is a `type` forced as in this PR or we are trying to load posts automatically and we loaded all old posts then we try loading new posts if the screen is not filled.

Reply:
Doesn't that allow a scenario where you scroll up (so `type` is `BEFORE_ID`) and it loads more posts after though?

Reply:
This method is called on two cases

1. if the screen is not filled with enough posts 
   * Try load previous posts
   * if all previous posts are loaded load newer posts and enable manual links if we are exceeding a certain number of API calls

2. Now we added the support of calling this if on load we see a loader close to bottom of the screen

 > Doesn't that allow a scenario where you scroll up (so `type` is `BEFORE_ID`) and it loads more posts after though?

So no we won't hit this issue because we wont be calling this on manual user scroll

Reply:
Okay, that makes sense. I misunderstood when this is supposed to be called.